lunavl/sdk/descriptors/descriptors.py

```diff
@@ -148,20 +151,20 @@ def __getitem__(self, i) -> FaceDescriptor:
         """
         if i >= len(self):
             raise IndexError(f"Descriptor index '{i}' out of range")  # todo remove after
-        descriptor = FaceDescriptor(self._coreEstimation.getDescriptorFast(i), self.scores[i])
+        descriptor = self.__class__._descriptorFactory(self._coreEstimation.getDescriptorFast(i), self.scores[i])
         return descriptor
 
-    def __iter__(self) -> Iterator[FaceDescriptor]:
+    def __iter__(self) -> Iterator[BaseDescriptor]:
         """
         Iterator by batch.
 
         Returns:
             iterator by descriptors.
         """
         for index in range(len(self)):
-            yield FaceDescriptor(self._coreEstimation.getDescriptorFast(index), self.scores[index])
+            yield self.__class__._descriptorFactory(self._coreEstimation.getDescriptorFast(index), self.scores[index])
```

Comment:
What for? Why won't work without black magic?
```suggestion
            yield self._descriptorFactory(self._coreEstimation.getDescriptorFast(index), self.scores[index])
```
Search for other `__class__` also pls.

Reply:
also no `__class__` usage [here](https://github.com/matemax/lunasdk/pull/82/files#diff-4892cfe567cd5f0b37e22f0c241672d7R269)